Unity/Assets/Cars/SteeringWheelSimCar/elements/RayCasts.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+public class RayCasts : MonoBehaviour
+{
+    public float length = 10;
+
+    [Range(1, 360)] public float angle = 120;
+
+    public int numberOfRays = 10;
+
+    public bool drawRays = false;
+
+    public float rayCastsElevation = 0.4f;
+    private void FixedUpdate()
+    {
+        if(drawRays)
+            DrawRayCasts();
+    }
+
+    private void DrawRayCasts()
+    {
+        RaycastHit hit;
+        var rayInFront = ((transform.rotation) * Vector3.forward);
+        for (var j = 0 - angle / 2; j < angle / 2; j += angle / numberOfRays)
+        {
+            var rayVector = Quaternion.AngleAxis(j, Vector3.up) * rayInFront;
+            var position = transform.position + new Vector3(0, rayCastsElevation, 0);
+
+            var ray = new Ray(position, rayVector);
+            if (Physics.Raycast(ray, out hit, length))
+            {
+                Debug.DrawRay(position, rayVector * hit.distance, Color.yellow, 0, true);
+                //print("distance: " + hit.distance + "angle: " + j);
+            }
+            else
+            {
+                Debug.DrawRay(position, rayVector * length, Color.white, 0, true);
+            }
+        }
+    }
+
+    public RayCastObservations GetHitsAndDistances()
```

Comment:
"GetRayCastObservations" for greater consistency

Reply:
👍